src/components/Categories.tsx

```diff
@@ -0,0 +1,266 @@
+import { paddings } from "../styles/branding";
+import { primary } from "../styles/styleGuide";
+import { CategoryDetailsNavigationProps } from "../navigation/NavigationTypes";
+import { useNavigation } from "@react-navigation/native";
+import Loading from "./Loading";
+import { useTranslation } from "react-i18next";
+import mealService from "../services";
+
+const HomeScreenCategories = () => {
+  const navigation = useNavigation<CategoryDetailsNavigationProps>();
+  const [categories, onChangeCategories] = useState<FetchCategoriesResponse>();
+  const [isLoading, setLoading] = useState<boolean>(false);
+  const { t } = useTranslation();
+  const { height } = useWindowDimensions();
+
+  const onCategorySelected = (category: Category) => {
+    navigation.navigate("CategoryDetails", {
+      categoryName: category.strCategory,
+    });
+  };
+
+  interface RenderCategoryItem {
+    item: Category;
+    index: number;
+  }
+
+  const renderItem = ({ item, index }: RenderCategoryItem) => {
+    const isItemPositionEven = index % 2 === 0;
+    return (
+      <TouchableOpacity
+        activeOpacity={0.8}
+        onPress={() => {
+          onCategorySelected(item);
+        }}
+        style={[
+          styles.container,
+          index == 0 ? styles.firstListElementStyle : null,
+        ]}
+      >
+        <View
+          style={[
+            styles.innerViewStyle,
+            isItemPositionEven
+              ? styles.evenListElementStyle
+              : styles.oddListElementStyle,
+          ]}
+        >
+          <View style={styles.centerAbsoluteStyle}>
+            <Image
+              source={require("../../assets/images/background_image.png")}
+              style={[
+                isItemPositionEven
+                  ? styles.evenPositionListElementStyle
+                  : styles.oddPositionListElementStyle,
+                styles.categoryBackgroundStyle,
+              ]}
+            />
+          </View>
+          <View style={styles.textItemViewStyle}>
+            <Text style={styles.text}>{item.strCategory}</Text>
+          </View>
+          <View style={styles.imageContainer}>
+            <Image
+              source={{ uri: item.strCategoryThumb }}
+              style={styles.image}
+            />
+          </View>
+        </View>
+      </TouchableOpacity>
+    );
+  };
+
+  const getCategories = async () => {
+    try {
+      setLoading(true);
+      const response = await mealService.fetchCategoriesApi();
+      onChangeCategories(response);
+    } catch (e) {
+      console.error("[ CATEGORIES]", e);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    getCategories();
+  }, []);
+
+  return (
+    <View style={styles.mainViewStyle}>
+      {isLoading ? (
+        <Loading style={styles.loadingStyle} />
+      ) : (
+        <FlatList<Category>
+          ListHeaderComponent={() => {
+            return (
+              <View
```

Comment:
I think you can create a component with this

Reply:
I already create one while implementing the next features. This change will be visible in next pull requests.